### PR TITLE
Release report maintenance and Microbenchmark UI change

### PR DESCRIPTION
**Pull request by boshek**
This PR does a bit of maintenance on the release report including:
- bump the version of conbenchcoms
- properly facet macrobenchmarks by cleaning up label

This report is tested locally by comparing this baseline https://github.com/apache/arrow/commit/2dcee3f82c6cf54b53a64729fd81840efa583244 to this contender  https://github.com/apache/arrow/commit/f60c2818092a407eddfaaacadac775a14e0d28d1

This PR also re-enables changes previously reverted in #73. 

FYI @assignUser and @raulcd

### CODE REVIEW

**Comment by assignUser** [comment]
Nice! Could you add those commits into the workflow so we don't just get the empty page with a warning as the artifact?

File: .github/workflows/performance-release-report.yml [hunk 1]
```diff
@@ -99,3 +99,9 @@ jobs:
           BUCKET: ${{ secrets.CROSSBOW_DOCS_S3_BUCKET }}
         run: |
           aws s3 cp performance-release-report.html $BUCKET/release_reports/arrow-release-report-${{ env.RC_LABEL }}.html
+      
+      - name: Create html artifact
+        uses: actions/upload-artifact@v4
+        with:
+          name: Create html artifact
+          path: performance-release-report/perormance-release-report.html
```
Comment (assignUser) [inline .github/workflows/performance-release-report.yml:107]:
```suggestion
```
That all ready happens in the step before :)

**Review by assignUser — APPROVED** [review]

**Comment by assignUser** [comment]
Looking at the rendered report a verbatime html thing seems to have  snuck in:
image
Also visible when something is selected. Btw why re-revert of #73? (probably a discussion happened offline?)
image

**Comment by boshek** [comment]
> Looking at the rendered report a verbatime html thing seems to have snuck in: image Also visible when something is selected. Btw why re-revert of #73? (probably a discussion happened offline?) image

Ah good catch thanks. Should be good to go now.

**Review by assignUser — APPROVED** [review]
:+1:

**Pull request merged**